in-progress/8754-slashing-and staking.md

```diff
@@ -0,0 +1,357 @@
+  function widthdraw(address rollupAddress, address validatorAddress) external returns(bool);
+  function unstakeValidators(addresss rollupAddress, address validatorAddress) external returns(bool);
+  function slashValidator(address rollupAddress, address validatorAddress) external returns(bool);
+  function getRollup() internal view returns(address);
+}
+```
+
+### Deposit Function
+
+* Validators must deposit at least `MIN_DEPOSIT_AMOUNT`.
+* A successful call results in an entry in the `Deposits` mapping. This mapping is a `mapping(address rollupAddress => mapping(address validatorAddress => DepositLib.DepositObject))`
+* A `DepositObject` contains references to the `withdrawalAddress` and `followingRegistryBool` variables. It is unique per rollupAddress / validatorAddress combination. 
+* Therefore validators can have multiple deposits corresponding to different Rollups. But for each Rollup, they can only have one deposited balance, one withdrawal address and one `followingRegistryBool` flag. 
+* Once a validator deposit for a particular Rollup meets or exceeds `MIN_ACTIVATION_AMOUNT`, the Rollup instance can add them to the validator set by calling `activateValidator`
+* Any amounts specified by `activateValidator` are removed from `Deposits` and accounted for in a different mapping, say `StakedDeposits`.
+* Amounts in `StakedDeposits`are now subject to withdrawal delays and slashing requirements. 
+
+### Entering the Validator Set
+
+Each Rollup implements its own logic for how validators join the validator set. I expect standard delays (i.e. 10 epochs) before a validator actually joins the set. 
+
+Once the Rollup instance calls `activateValidator(_rollupAddress, _valAddress, _amount)`, 
+
+```solidity
+ // ...
+Deposits[_rollupAddress][_valAddress].amount -= _amount
+StakedDeposits[_rollupAddress][_valAddress].amount += _amount // if it exists, otherwise create it
+ // ...
+```
+In visual form: 
+
+```mermaid
+sequenceDiagram
+title Deposit and Enter the Validator Set
+
+    participant A as Validator
+    participant C as Deposits Contract
+    participant D as Registry
+    participant B as Rollup Instance
+
+    A ->> C:  deposit(amount, followingRegistry=true)
+    C ->> D:  registry.getRollup()
+    C ->> C:  Create DepositObject in Deposits[getRollup()][validatorAddress]
+    A ->> B:  joinValidator()
+    B ->> C:  Check balance of validator
+
+    alt If they have MIN_ACTIVATION_BALANCE
+        B ->> C: Make deposit at stake i.e. call activateValidator()
+        B ->> B: Add to validator set
+    else If they don't
+        B ->> B: Do nothing
+    end
+```
+
+### Migrating Stake
+
+The `Deposit` contract is owned by the Governance Contract. As such in the event of an upgrade, the Apella can change the `StakedDeposits` mapping so that any `StakeObject` with `followingRegistryBool=true` can be moved to the new Rollup as pointed to by the Registry contract. 
```

Comment:
May want to clarify that this is only an upgrade in the canonical rollup version, not any governance proposal.